### ISSUE TITLE
ARM GNU Toolchain 13.2 

### ISSUE THREAD

**Issue by JeremyMorlier** [iml130/iree-bare-metal-arm#306]
Issue when compiling with [ARM GUN Toolchain 13.2 rel1](https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads)
Compiling options :
```export PATH_TO_ARM_TOOLCHAIN="PATH/Toolchain/arm-gnu-toolchain-13.2.rel1-x86_64-arm-none-eabi"
export CUSTOM_ARM_LINKER_FLAGS=""
export PATH_TO_LINKER_SCRIPT="`realpath ../build_tools/stm32f407xg-cmsis.ld`"
export PATH_TO_IREE_HOST_BIN_DIR="`realpath ../build-iree-host-tools/bin`"
cmake -GNinja \
      -DBUILD_WITH_CMSIS=ON \
      -DCMAKE_TOOLCHAIN_FILE="`realpath ../build_tools/cmake/arm-none-eabi-gcc.cmake`" \
      -DARM_TOOLCHAIN_ROOT="${PATH_TO_ARM_TOOLCHAIN}" \
      -DARM_CPU="cortex-m4" \
      -DARM_TARGET="STM32F407" \
      -DIREE_ERROR_ON_MISSING_SUBMODULES=OFF \
      -DIREE_HAL_DRIVER_LOCAL_SYNC=ON \
      -DIREE_HAL_EXECUTABLE_LOADER_EMBEDDED_ELF=ON \
      -DIREE_HAL_EXECUTABLE_LOADER_VMVX_MODULE=ON \
      -DIREE_HOST_BIN_DIR="${PATH_TO_IREE_HOST_BIN_DIR}" \
      -DCUSTOM_ARM_LINKER_FLAGS="${CUSTOM_ARM_LINKER_FLAGS}" \
      -DLINKER_SCRIPT="${PATH_TO_LINKER_SCRIPT}" \
      -DUSE_UART2=ON \
      ..
```
log : 
```[0/2] Re-checking globbed directories...
[3/129] Building C object third_party/iree/runtim...ns_ukernel_ukernel_noweak.objects.dir/mmt4d.c.obj
FAILED: third_party/iree/runtime/src/iree/builtins/ukernel/CMakeFiles/iree_builtins_ukernel_ukernel_noweak.objects.dir/mmt4d.c.obj 
PATH/Toolchain/arm-gnu-toolchain-13.2.Rel1-x86_64-arm-none-eabi/bin/arm-none-eabi-gcc -DFLATCC_USE_GENERIC_ALIGNED_ALLOC -DIREE_PLATFORM_GENERIC -DIREE_STATUS_FEATURES=0 -DIREE_SYNCHRONIZATION_DISABLE_UNSAFE=1 -isystem PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree -isystem PATH/Documents/MLIR/iree-bare-metal-arm/build/third_party/iree -isystem PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/.. -isystem PATH/Documents/MLIR/iree-bare-metal-arm/build/third_party/iree/runtime/src/iree/.. -mthumb -march=armv7e-m -mfloat-abi=hard -mfpu=fpv4-sp-d16 -DIREE_TIME_NOW_FN="{ return 0; }" -DIREE_WAIT_UNTIL_FN=wait_until -Wl,--gc-sections -ffunction-sections -fdata-sections  -Wno-char-subscripts -Wno-format -Wno-implicit-function-declaration -Wno-unused-variable -fPIC -fvisibility=hidden -Wall -Werror -Wno-address-of-packed-member -Wno-comment -Wno-format-zero-length -Wno-uninitialized -Wno-pointer-sign -Wno-sign-compare -Wno-unused-function -fmacro-prefix-map=PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree=iree -std=gnu11 -MD -MT third_party/iree/runtime/src/iree/builtins/ukernel/CMakeFiles/iree_builtins_ukernel_ukernel_noweak.objects.dir/mmt4d.c.obj -MF third_party/iree/runtime/src/iree/builtins/ukernel/CMakeFiles/iree_builtins_ukernel_ukernel_noweak.objects.dir/mmt4d.c.obj.d -o third_party/iree/runtime/src/iree/builtins/ukernel/CMakeFiles/iree_builtins_ukernel_ukernel_noweak.objects.dir/mmt4d.c.obj -c PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/mmt4d.c
In file included from PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/mmt4d.h:10,
                 from PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/mmt4d.c:7:
PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/common.h: In function 'iree_uk_f16_to_f32':
PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/common.h:791:3: error: '_Float16' is not supported on this target
  791 |   _Float16 builtin_float16_value;
      |   ^~~~~~~~
PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/common.h: In function 'iree_uk_f32_to_f16':
PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/common.h:803:3: error: '_Float16' is not supported on this target
  803 |   _Float16 builtin_float16_value = (_Float16)value;
      |   ^~~~~~~~
PATH/Documents/MLIR/iree-bare-metal-arm/third_party/iree/runtime/src/iree/builtins/ukernel/common.h:803:37: error: '_Float16' is not supported on this target
  803 |   _Float16 builtin_float16_value = (_Float16)value;
      | ```

**Comment by marbre** [comment]
I am currently looking (again) into bumping the IREE submodule before upgrading the Arm toolchain. We started working on the integrate last year but are facing some issue. Sorry for the delay but we're on it.

**Comment by marbre** [comment]
The issue is resolved with #309 / 4d2d2ac and the toolchain can finally be upgraded to version 13.2.Rel1 (we will do with #312).
